src/fsharp/MethodCalls.fs

```diff
@@ -42,13 +43,18 @@ open FSharp.Compiler.ExtensionTyping
 /// expression, e.g. a lambda expression may be converted to a delegate as 
 /// an adhoc conversion. 
 ///
-/// The bool indicates if named using a '?' 
+/// The bool indicates if named using a '?', making the caller argument explicit-optional
 type CallerArg<'T> = 
+
     /// CallerArg(ty, range, isOpt, exprInfo)
     | CallerArg of TType * range * bool * 'T  
-    member x.Type = (let (CallerArg(ty, _, _, _)) = x in ty)
+
+    member x.CallerArgumentType = (let (CallerArg(ty, _, _, _)) = x in ty)
```

Comment:
nit: The type is called `CallerArg` and this property on it is called `CallerArgumentType` which feels a little redundant, I feel `Type` was more appropriate.

Reply:
We generally qualify the name "type", e.g. "PropertyType" on "PropertyInfo", "FieldType" on "FieldInfo" etc.    I think it's good practice given how many meanings the word carries

Reply:
FYI in this PR https://github.com/dotnet/fsharp/pull/6596/files#diff-456bef6dcdc9277074038269046cab70 I'm changing this type to a record, it didn't impact too much code.

I also introduce `CallerArgs` to replace `curriedCallerArgs: (CallerArg<'T> list * CallerNamedArg<'T> list) list,` in `type CalledMeth<'T>` constructor.

I'd like to understand why those values end up being list of tuple of lists instead of just a tuple of lists? It seems to run pretty deep up to `SynValInfo` but there is no indication of what those nested lists are.

@dsyme: could you cast some light?

Reply:
@dsyme ok, will keep that in mind. :)